### ISSUE TITLE
addon.storage API

### ISSUE THREAD

**Issue by TheColaber** [ScratchAddons/ScratchAddons#1653]
Would be nice for future addons...

**Comment by RedGuy12** [comment]
What would it be?

**Comment by retronbv** [comment]
> What would it be?

Like localstorage for addons?

**Comment by RedGuy12** [comment]
But local storage and session storage already have great api's... maybe for cookies but addons barely use cookies...
________________________________

**Comment by retronbv** [comment]
> But local storage and session storage already have great api's... maybe for cookies but addons barely use cookies...
> […](#)

No, localstorage FOR each addon

**Comment by pufferfish101007** [comment]
> But local storage and session storage already have great api's... maybe for cookies but addons barely use cookies...

The point is addons can't use that reliably, as it could be cleared for multiple reasons, and if you want the storage to be accessible from popups as well as scratch that won't work, and if we choose to add more sites that addons work on in the future that will work even less. Therefore it would be useful to have a separate addon.storage api.

**Comment by TheColaber** [comment]
> The point is addons can't use that reliably, as it could be cleared for multiple reasons, and if you want the storage to be accessible from popups as well as scratch that won't work, and if we choose to add more sites that addons work on in the future that will work even less. Therefore it would be useful to have a separate addon.storage API.

Correct. Using cookies and local storage is not reliable as scratch and other extensions and clears it if they wanted to. Also, this restiricks us to be limited to only using scratch to use addons on. If we wanted to allow more mods, this wouldn't be possible. Also, if we wanted webpages (like the popup) to use this data, this could only work by interaction with the chrome.storage API, which is exactly what we need. Implementing dynamic-length settings is something we have not implemented and is in the process, however they're also other reasons why storage might be needed.
